### PR TITLE
DBZ-1824 Restart incomplete Tx for wal2json

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-1824

The fix goes against processing heartbeat logic - it has to be reconciled with it yet.

### CODE REVIEW

**Comment by jpechane** [comment]
I wonder if the problem hit related to dupplicated LSNs is related to https://github.com/debezium/debezium/blob/365eb0d76e7b73e70e09fcc798100dcc106304ff/debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/pgoutput/PgOutputMessageDecoder.java#L133-L138

@Naros Any idea, does it look familiar?

**Comment by Naros** [comment]
Hi @jpechane, it does but since we did not need the `COMMIT` messages for anything at the time, I did not spend too many cycles investigating why I observed.  If you do not skip the commit message in the decoder, does the fail continue to fail?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java [hunk 1]
```diff
@@ -402,6 +402,16 @@ public void read(ReplicationMessageProcessor processor) throws SQLException, Int
             public boolean readPending(ReplicationMessageProcessor processor) throws SQLException, InterruptedException {
                 ByteBuffer read = stream.readPending();
                 final long lastReceiveLsn = stream.getLastReceiveLSN().asLong();
+                // Conflict between DBZ-1824 and DBZ-800
+                // PostgreSQL replication stream increases last received LSN even if no message arrives
+                // Then if data message arrives it will have the same LSN as the updated one.
+                // For wal2json it means that the transaction in progress will contain last_processed_lsn to the same value
+                // as transaction LSN and upon restart the incomplete transaction will be skipped instead of being replayed
+                // Also there is an open question whether this can lead to premature LSN flushing as if empty message and data
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java]: That sounds a bit concerning. Are you planning to follow up on this? Anything we should put into the docs?

**Comment by jpechane** [comment]
@gunnarmorling Update to the latest master and the changes in the postgres message processing

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java [hunk 2]
```diff
@@ -129,7 +129,9 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
                     if (message.isTransactionalMessage() || message.getOperation() == Operation.NOOP) {
                         if (!connectorConfig.shouldProvideTransactionMetadata()) {
                             LOGGER.trace("Received transactional message {}", message);
-                            skipMessage(lsn);
+                            if (message.getOperation() == Operation.COMMIT) {
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java]: Can you add a comment why only on COMMIT it's skipped?

**Comment by jpechane** [comment]
@gunnarmorling Finally passing

**Comment by gunnarmorling** [comment]
And applied. Thanks, @jpechane!

**Pull request merged**